### ISSUE TITLE
Multiple players may have the same name.

### ISSUE THREAD

**Issue by DavidCo113** [SpadesX/SpadesX#72]
**Describe the bug**
In the event that any player has a name (depending on player ID length) 15 or 14 characters in length, another player may freely copy that name, with any of the following effects:
 - Nothing to distinguish the two names whatsoever.
 - (70025abf4472ea4781b964477a3e79530b6fdcfd) The full player ID is concatenated, while making the player name too long in the process.
 - A single character of the player ID is concatenated, (70025abf4472ea4781b964477a3e79530b6fdcfd) resulting in a player name which is too long.

Anything with (70025abf4472ea4781b964477a3e79530b6fdcfd) and not (https://github.com/SpadesX/SpadesX/commit/75bf890639e68df60a81d61932cd9b112c7b7379) preceding it is fixed with #71. If the player name is less than 15 or 14 characters, the player which copied the name is correctly distinguished from the original by concatenating its player ID to the name.

**To Reproduce**
1. Add a player to the game with a name which is 15 characters in length.
2. Add another one with the same name.

**Expected behavior**
Player names which are not identical (70025abf4472ea4781b964477a3e79530b6fdcfd) or too large.

**Screenshots**
(70025abf4472ea4781b964477a3e79530b6fdcfd) Player names that are 15 characters in length, with the copied name being one character too long.
![image](https://github.com/SpadesX/SpadesX/assets/36682328/3893cad2-ddd0-4239-91d7-cfaf399ee2a9)

(https://github.com/SpadesX/SpadesX/commit/75bf890639e68df60a81d61932cd9b112c7b7379) Player names that are 15 characters in length, and therefore indistinguishable.
![image](https://github.com/SpadesX/SpadesX/assets/36682328/749497bd-9dbf-45a5-a122-e9b17bf60932)

(70025abf4472ea4781b964477a3e79530b6fdcfd/https://github.com/SpadesX/SpadesX/commit/75bf890639e68df60a81d61932cd9b112c7b7379) Properly distinguished player names.
![image](https://github.com/SpadesX/SpadesX/assets/36682328/132cb144-cdd9-4317-ae7f-23959c6c89ce)

**Server (please complete the following information):**
 - OS: Arch (GNU/)Linux
 - Version: Variable, see #71.

**Comment by DavidCo113** [comment]
I'll go ahead and de-clutter this a little now that 75bf890639e68df60a81d61932cd9b112c7b7379 has been merged.

**Comment by DavidCo113** [comment]
Although, then again, that would make half of the purpose of #72 missing, so maybe not.

**Comment by lokka30** [comment]
For a bit of fun I've written this up, food for thought

<details>
<summary>Click to show/hide</summary>

```c
// Created by lokka30 on 2024-02-20 11:19 AM GMT+8:00

#include <stdio.h>
#include <string.h>
#include <stdbool.h>

bool is_duplicate_player_name(char *name) {
    // pretends that there are 11 checks required before a non-duplicate is found
    static int name_checks_remaining = 11;
    name_checks_remaining--;
    return name_checks_remaining != 0;
}

void update_name_if_duplicate(char* name) {
    char   buffer[17];
    size_t count = 1;

    bool is_duplicate = is_duplicate_player_name(name);

    if(!is_duplicate) {
        return;
    }

    while(is_duplicate) {
        char   count_str[4];
        size_t count_strlen = snprintf(count_str, 4, "%lu", count);
        size_t name_strlen = strlen(name);
        size_t trunc_name_len = (name_strlen + count_strlen > 16) ? name_strlen - count_strlen : name_strlen;

        strcpy(buffer, name);

        snprintf(buffer, 17, "%.*s%lu", (int) trunc_name_len, name, count);

        is_duplicate = is_duplicate_player_name(buffer);
        count++;
    }
    strcpy(name, buffer);
}

void main() {
    char name[17];

    strcpy(name, "thequickbrownfox");
    // strcpy(name, "h");
    // strcpy(name, "he");
    // strcpy(name, "hello");

    update_name_if_duplicate(name);
    return 0;
}
```

</details>

**Comment by lokka30** [comment]
Hi @DavidCo113, are you still working on a fix for this?

**Comment by DavidCo113** [comment]
> Hi @DavidCo113, are you still working on a fix for this?
I am not.

**Comment by lokka30** [comment]
I think I'm onto a working solution, I'll clean up the code and work on a PR.

<img width="1440" alt="image" src="https://github.com/user-attachments/assets/805f4bea-5775-482b-a2ba-064d16aef540">

<img width="257" alt="image" src="https://github.com/user-attachments/assets/ba3577d4-e6bd-42aa-9b56-96d429dd789a">

**Comment by lokka30** [comment]
Just to update, my attempt to fix this is in #86 and is ready for review.
